src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -2378,9 +2380,10 @@ trait Types
     }
     //OPT specialize equals
     override final def equals(other: Any): Boolean = {
-      other match {
+      if (this eq other.asInstanceOf[AnyRef]) true
+      else other match {
         case otherTypeRef: TypeRef =>
-          pre.equals(otherTypeRef.pre) && sym.eq(otherTypeRef.sym) && sameElementsEquals(args, otherTypeRef.args)
+          Objects.equals(pre, otherTypeRef.pre) && sym.eq(otherTypeRef.sym) && sameElementsEquals(args, otherTypeRef.args)
```

Comment:
what's the motivation for using `Objects.equals`? are there cases where the prefix is `null`?

Reply:
It has a fast path based on reference equality before performing the virtual call to `Type.equals`.